### PR TITLE
Overhaul README.md

### PR DESCRIPTION
**Pull request by dune73**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,23 +34,38 @@ go install github.com/coreruleset/go-ftw@latest
...
 
 ### YAML Config file
 
-With a configuration file you can alter the test results, set paths for your environment, or enable features. The config file has four basic values:
+With the configuration set paths for your environment, enable and disabled features and you can also use it to alter the test results.
+
+The config file has four basic values:
+
+* `logfile` : path to WAF log with alert messages, relative of absolute
+* `logmarkerheadername` : name of a HTTP header used for marking log messages, usually `X-CRS-TEST` (see section "How log parsing works" below)
```
Comment (fzipi) [inline README.md:53]:
```suggestion
* `logmarkerheadername` : name of a HTTP header used for marking log messages, usually `X-CRS-TEST` (see [How log parsing works](https://github.com/coreruleset/go-ftw#how-log-parsing-works) below)
```

File: README.md [hunk 2]
```diff
@@ -34,23 +34,38 @@ go install github.com/coreruleset/go-ftw@latest
...
-1. a WAF (doh!)
-2. a file where the waf stores the logs
-3. a config file, or environment variables, with the information to get the logs and how to parse them (I might embed this for the most commonly used, like Apache/NGiNX), and additional options described next.
+The go-ftw is designed to run Web Application Firewall (WAF) unit tests. The primary focus is the [OWASP ModSecurity Core Rule Set](https://github.com/coreruleset/coreruleset).
+
+In order to run the tests, you need to prepare the following:
+
+1. Active WAF
+2. Log where the WAF writes the alert messages
+3. go-ftw config file `.ftw.yaml` in the local folder or in your home folder (see section "YAML Config file" for more information.
```
Comment (fzipi) [inline README.md:43]:
```suggestion
3. go-ftw config file `.ftw.yaml` in the local folder or in your home folder (see [YAML Config file](https://github.com/coreruleset/go-ftw#yaml-config-file) for more information).
```

File: README.md [hunk 3]
```diff
@@ -34,23 +34,38 @@ go install github.com/coreruleset/go-ftw@latest
...
 
 ```yaml
-logfile: <the relative path to the WAF logfile>
-logmarkerheadername: <a header name used for log parsing (see "How log parsing works" below)>
-testoverride: <a list of things to override (see "Overriding tests" below)>
-mode: "default" or "cloud" (only change it if you need "cloud")
+logfile: /apache/logs/error.log
+logmarkerheadername: X-CRS-TEST
+testoverride:
+mode: "default"
```
Comment (fzipi) [inline README.md:64]: Maybe we want to add more examples:
- this has mostly everything set, with the logfile being absolute
- add a relative logfile example
- add a minimal config: the bare minimum needed is the `logfile`.

File: README.md [hunk 4]
```diff
@@ -226,33 +241,34 @@ mode: 'cloud'
...
-you will need to instrument it with the same idea (unless you are using "cloud mode").
+the `CRS_ENABLE_TEST_MARKER` environment variable. If you are testing a different test setup, you will need to instrument it with a rule that generated the marker in the log file via a rule alert (unless you are using "cloud mode").
+
 The rule for CRS looks like this:
+
 ```
 # Write the value from the X-CRS-Test header as a marker to the log
 SecRule REQUEST_HEADERS:X-CRS-Test "@rx ^.*$" \
-  "id:999999,\
+  "id:900910,\
```
Comment (fzipi) [inline README.md:262]: If you change it here, you need probably to change it all over the place. In particular, [here](https://github.com/coreruleset/modsecurity-crs-docker/blob/b0603c1832a38eb578eefce462218f367af7c88e/src/opt/modsecurity/activate-rules.sh#L133)

**Review by fzipi — CHANGES_REQUESTED** [review]
Looks good. For consistency, we need to make changes in additional places probably.

**Comment by dune73** [comment]
Thank you @fzipi. I think I have covered all your comments.

I'm not a fan of rule ID 999999. But it's not necessary to start to change a lot of things all of a sudden.

**Review by fzipi — APPROVED** [review]

**Pull request merged**